lib/chef/provider/cron.rb

```diff
@@ -89,11 +89,18 @@ def load_current_resource
       end
 
       def cron_different?
+        logger.warn("It is not idempotent to use `environment` for an entry that can be specified as a `property`") if property_with_environment?
```

Comment:
I found this error message confusing - it would be better if it was more explicit, such as "It looks like you are trying to set the MAILTO environment variable.  Please use the `mailto` resource property instead."